docs/source/troubleshooting.rst

```diff
@@ -0,0 +1,211 @@
+Troubleshooting Guide
+====================
+
+This guide helps you resolve common issues when using TACS. If you don't find your issue here, check the `GitHub Issues <https://github.com/smdogroup/tacs/issues>`_ for community support.
+
+Installation Issues
+-------------------
+
+**ImportError: No module named 'tacs'**
+
+*Cause*: TACS is not installed or not in the Python path.
+
+*Solutions*:
+
+- Ensure conda environment is activated: ``conda activate tacs``
+- Reinstall TACS: ``mamba install -c conda-forge -c smdogroup tacs``
+- For source installation, ensure Python interface is built: ``pip install -e .[all]``
+
+**ImportError: libtacs.so not found**
+
+*Cause*: TACS C++ library is not in the library path.
+
+*Solutions*:
+
+.. code-block:: bash
+
+   # Add TACS lib directory to LD_LIBRARY_PATH
+   export LD_LIBRARY_PATH=/path/to/tacs/lib:$LD_LIBRARY_PATH
+   
+   # Or add to your .bashrc for permanent fix
+   echo 'export LD_LIBRARY_PATH=/path/to/tacs/lib:$LD_LIBRARY_PATH' >> ~/.bashrc
+
+**MPI compilation errors**
+
+*Cause*: MPI compiler not found or misconfigured.
+
+*Solutions*:
+
+- Ensure MPI is installed: ``which mpicc``
+- Check MPI compiler in Makefile.in: ``CXX = mpicxx``
+- For Intel MPI: ``CXX = icpc -lmpi``
+
+Analysis Issues
+---------------
+
+**Analysis fails to converge**
+
+*Cause*: Poor conditioning, incorrect boundary conditions, or excessive loads.
+
+*Solutions*:
+
+- Check boundary conditions are properly applied
+- Verify loads are reasonable for the structure
+- Ensure material properties are correct
+- Try reducing load magnitude
+- Check for rigid body modes
+
+**Element callback errors**
+
+*Cause*: Mismatch between BDF element types and callback function.
+
+*Solutions*:
+
+- Verify element types in BDF file match callback function
+- Add error handling for unexpected element types:
+
+  .. code-block:: python
+
+     for descript in elemDescripts:
+         if descript == 'CQUAD4':
+             elem = elements.Quad4Shell(transform, con)
+         else:
+             raise ValueError(f"Unexpected element type: {descript}")
+
+**Memory errors with large models**
+
+*Cause*: Insufficient memory for analysis or postprocessing.
+
+*Solutions*:
+
+- Consider using more powerful hardware
```

Comment:
Again, is the advice "use more powerful hardware if you run out of memory" really useful? It would be much more useful to explain what output you'll see if you ran out of memory and if there are any options you can set in TACS to reduce memory usage